src/v/cluster/shard_placement_table.h

```diff
@@ -104,9 +104,10 @@ class shard_placement_table
           : target(target)
           , shard_revision(rev) {}
 
-        /// If this shard is the initial shard for this partition on this node,
-        /// this field will contain the corresponding shard revision.
-        model::shard_revision_id _is_initial_at_revision;
+        /// If this shard is the initial shard for some incarnation of this
+        /// partition on this node, this field will contain the corresponding
+        /// log revision.
+        std::optional<model::revision_id> _is_initial_for;
```

Comment:
Why is tracking the initial shard important? Is it because some residual state sticks around on the initial shard instead? I didn't look at the RFC is this explained there? I'll go take a look at that.

Reply:
Well, we have to remember that the initial shard can create the partition without waiting for anybody. But we can't do it by immediately setting the current state to "hosted", because it can still be occupied by the previous log revision. So `_is_initial_for` becomes an intermediate place where we can store this information while the previous log revision is being cleaned up.